### ISSUE TITLE
gf server error in BLAT

### ISSUE THREAD

**Issue by JAYRJPT** [stjude/CICERO#19]
I have installed Cicero in my computer but I am facing issue in running it.
my command 
**/media/deepak/EXTRA/CICERO-master/src/scripts/Cicero.sh -n 20 -b /media/deepak/Deepak4T/NEW_SAMPLE/SAMPLE_3/star/sample.sorted.bam -g /media/deepak/Deepak4T/Genomedir/hg38/hg38.fa -r /home/deepak/Downloads/reference**

output..
Optimize: setting SC_SHIFT=10, SC_CUTOFF=3, THRESHOLD=200000
/media/deepak/EXTRA/CICERO-master/configs
configs.ngs.26697.tmp
SJ_CONFIGS=configs.ngs.26697.tmp
awk: fatal: cannot open file `configs.ngs.26697.tmp/genome//media/deepak/Deepak4T/Genomedir/hg38/hg38.fa.config.txt' for reading (No such file or directory)
awk: fatal: cannot open file `configs.ngs.26697.tmp/genome//media/deepak/Deepak4T/Genomedir/hg38/hg38.fa.config.txt' for reading (No such file or directory)
awk: fatal: cannot open file `configs.ngs.26697.tmp/genome//media/deepak/Deepak4T/Genomedir/hg38/hg38.fa.config.txt' for reading (No such file or directory)
awk: fatal: cannot open file `configs.ngs.26697.tmp/genome//media/deepak/Deepak4T/Genomedir/hg38/hg38.fa.config.txt' for reading (No such file or directory)
Starting local blat server
Blat server not yet running ...
ERROR: The server has probably died. Please review error logs at below location:
/media/deepak/EXTRA/CICERO-master/src/scripts/gfServer.ngs.26697.err

KIndly help to resolve my issue.

**Comment by adthrasher** [comment]
Unfortunately, CICERO does not currently support hg38. The "-g" parameter expects a string, the only currently valid option is "GRCh37-lite".

**Comment by JAYRJPT** [comment]
Thanks for the reply.
Even after changing the "-g" parameter to "GRCh37-lite" , I am getting the same error.
I think it is the problem of BLAT.
Please suggest how to resolve that issue.

Thanks and Regards

**Comment by adthrasher** [comment]
Can you share the full error once you've changed the genome argument? Can you also share the indicated log file from BLAT?

**Comment by Miachol** [comment]
@JAYRJPT Maybe you need to create and modify the config file `CICERO/configs/genome/GRCh37-lite.config.txt`, default template is `CICERO/configs/genome/GRCh37-lite.template.txt`.

**Comment by JAYRJPT** [comment]
command used-
**/media/deepak/EXTRA/CICERO-master/src/scripts/Cicero.sh -n 20 -b /media/deepak/Deepak4T/NEW_SAMPLE/SAMPLE_3/star/sample.sorted.bam -g /home/deepak/Downloads/reference/Homo_sapiens/GRCh37-lite/2BIT/GRCh37-lite.2bit -r /home/deepak/Downloads/reference**

**Error-**
Optimize: setting SC_SHIFT=10, SC_CUTOFF=3, THRESHOLD=200000
/media/deepak/EXTRA/CICERO-master/configs
configs.ngs.36980.tmp
SJ_CONFIGS=configs.ngs.36980.tmp
awk: fatal: cannot open file `configs.ngs.36980.tmp/genome//home/deepak/Downloads/reference/Homo_sapiens/GRCh37-lite/2BIT/GRCh37-lite.2bit.config.txt' for reading (No such file or directory)
awk: fatal: cannot open file `configs.ngs.36980.tmp/genome//home/deepak/Downloads/reference/Homo_sapiens/GRCh37-lite/2BIT/GRCh37-lite.2bit.config.txt' for reading (No such file or directory)
awk: fatal: cannot open file `configs.ngs.36980.tmp/genome//home/deepak/Downloads/reference/Homo_sapiens/GRCh37-lite/2BIT/GRCh37-lite.2bit.config.txt' for reading (No such file or directory)
awk: fatal: cannot open file `configs.ngs.36980.tmp/genome//home/deepak/Downloads/reference/Homo_sapiens/GRCh37-lite/2BIT/GRCh37-lite.2bit.config.txt' for reading (No such file or directory)
Starting local blat server
Blat server not yet running ...
ERROR: The server has probably died. Please review error logs at below location:
/media/deepak/EXTRA/CICERO-master/src/scripts/gfServer.ngs.36980.err

**Log file** 
2020/06/13 17:23:25: error: gfServer v 35 - Make a server to quickly find where DNA occurs in genome.
To set up a server:
   gfServer start host port file(s)
   Where the files are in .nib or .2bit format
To remove a server:
   gfServer stop host port
To query a server with DNA sequence:
   gfServer query host port probe.fa
To query a server with protein sequence:
   gfServer protQuery host port probe.fa
To query a server with translated dna sequence:
   gfServer transQuery host port probe.fa
To query server with PCR primers
   gfServer pcr host port fPrimer rPrimer maxDistance
To process one probe fa file against a .nib format genome (not starting server):
   gfServer direct probe.fa file(s).nib
To test pcr without starting server:
   gfServer pcrDirect fPrimer rPrimer file(s).nib
To figure out usage level
   gfServer status host port
To get input file list
   gfServer files host port
Options:
   -tileSize=N size of n-mers to index.  Default is 11 for nucleotides, 4 for
               proteins (or translated nucleotides).
   -stepSize=N spacing between tiles. Default is tileSize.
   -minMatch=N Number of n-mer matches that trigger detailed alignment
               Default is 2 for nucleotides, 3 for protiens.
   -maxGap=N   Number of insertions or deletions allowed between n-mers.
               Default is 2 for nucleotides, 0 for protiens.
   -trans  Translate database to protein in 6 frames.  Note: it is best
           to run this on RepeatMasked data in this case.
   -log=logFile keep a log file that records server requests.
   -seqLog    Include sequences in log file (not logged with -syslog)
   -ipLog     Include user's IP in log file (not logged with -syslog)
   -syslog    Log to syslog
   -logFacility=facility log to the specified syslog facility - default local0.
   -mask      Use masking from nib file.
   -repMatch=N Number of occurrences of a tile (nmer) that trigger repeat masking the tile.
               Default is 2252.
   -maxDnaHits=N Maximum number of hits for a dna query that are sent from the server.
               Default is 100.
   -maxTransHits=N Maximum number of hits for a translated query that are sent from the server.
               Default is 200.
   -maxNtSize=N Maximum size of untranslated DNA query sequence
               Default is 40000
   -maxAaSize=N Maximum size of protein or translated DNA queries
               Default is 8000
   -canStop If set then a quit message will actually take down the
            server

2020/06/13 17:23:25: error: gfServer aborted

**Comment by JAYRJPT** [comment]
> @JAYRJPT Maybe you need to create and modify the config file `CICERO/configs/genome/GRCh37-lite.config.txt`, default template is `CICERO/configs/genome/GRCh37-lite.template.txt

I have checked the config file. It is " GRCh37-lite.template.txt"

**Comment by adthrasher** [comment]
The -g parameter requires a string, not a file. Try "-g GRCh37-lite". The code uses the config files along with the reference directory to find the 2BIT file.

**Comment by JAYRJPT** [comment]
Thanks for the explanation.
I am able to run the Cicero tool. However the application is struck at one point for a long time. Can you please suggest whether it is the part of the running process ?
The log is..
 **/media/deepak/EXTRA/CICERO-master/src/scripts/Cicero.sh -n 20 -b /media/deepak/Deepak4T/NEW_SAMPLE/SAMPLE_3/star/sample.sorted.bam -g GRCh37-lite -r /home/deepak/Downloads/reference**
Optimize: setting SC_SHIFT=10, SC_CUTOFF=3, THRESHOLD=200000
/media/deepak/EXTRA/CICERO-master/configs
configs.ngs.11332.tmp
SJ_CONFIGS=configs.ngs.11332.tmp
Starting local blat server
Blat server not yet running ...
Blat server not yet running ...
Blat server not yet running ...
Blat server not yet running ...
Blat server not yet running ...
Blat server not yet running ...
Blat server not yet running ...
Blat server not yet running ...
Blat server not yet running ...
Blat server not yet running ...
Blat server not yet running ...
Blat server not yet running ...
Blat server not yet running ...
Blat server not yet running ...
Blat server is up!
Step 01 - 2020.06.15 12:38:34 - ExtractSClips

The application is struck at step 1 since yesterday.

**Comment by adthrasher** [comment]
The first step extracts reads with soft clip positions. So the runtime will depend on the number of these events. The step runs in parallel over a number of regions. In the output directory, there should be a log file from [GNU parallel](https://www.gnu.org/software/parallel/) named "01_ExtractSClips.log". This should show the progress of the individual steps.

**Comment by Miachol** [comment]
I have tested the CICERO with '-n 39', and the longest case takes 3 and a half days to complete the whole workflow.

**Comment by JAYRJPT** [comment]
> _The first step extracts reads with soft clip positions. So the runtime will depend on the number of these events. The step runs in parallel over a number of regions. In the output directory, there should be a log file from [GNU parallel](https://www.gnu.org/software/parallel/) named "01_ExtractSClips.log". This should show the progress of the individual jobs._

I am sharing the 01_ExtractSClips.log and error file. I am getting some error therefore it is not moving forward.
**01_ExtractSClips.log**

Seq	Host	Starttime	JobRuntime	Send	Receive	Exitval	Signal	Command
1	:	1592379692.203	     0.000	  0	           0	         127	  0	          get_geneInfo.pl -i /media/deepak/Deepak4T/NEW_SAMPLE/SAMPLE_3/star/sample.sorted.bam -o CICERO_DATADIR/sample.sorted -l  -genome GRCh37-lite -s sample.sorted

**01_ExtractSClips.err**

/media/deepak/EXTRA/CICERO-master/src/scripts/Cicero.sh: line 240: getReadLength.sh: command not found
/media/deepak/EXTRA/CICERO-master/src/scripts/Cicero.sh: line 241: get_sc_cmds.pl: command not found
/bin/bash: get_geneInfo.pl: command not found

**Comment by adthrasher** [comment]
I notice you're calling Cicero.sh with the full path. Have you added the CICERO paths to your PATH and PERL5LIB environment variables? If not, bash will not know where to find those executables and libraries.

**Comment by JAYRJPT** [comment]
I have added the respective directory to the PATH and PER5LIB but still issue is not resolved.
I am getting the following error.
**01_ExtractSClips.err**
/home/deepak/software_library/bin/Cicero.sh: line 240: getReadLength.sh: command not found
/home/deepak/software_library/bin/Cicero.sh: line 241: get_sc_cmds.pl: command not found
Can't use 'defined(@array)' (Maybe you should just omit the defined()?) at /usr/local/bin/parallel line 120.

**Comment by JAYRJPT** [comment]
After adding "getReadLength.sh" and "get_sc_cmds.pl" to the PATH 
I am getting only the following error
Can't use 'defined(@array)' (Maybe you should just omit the defined()?) at /usr/local/bin/parallel line 120.
I think it is the error of GNU parallel.

**Comment by adthrasher** [comment]
Can you share the output of `parallel --version` and `perl --version`?

**Comment by JAYRJPT** [comment]
`parallel --version`
**Can't use 'defined(@array)' (Maybe you should just omit the defined()?) at /usr/local/bin/parallel line 120.**
`perl --version`

**This is perl 5, version 26, subversion 1 (v5.26.1) built for x86_64-linux-gnu-thread-multi
(with 67 registered patches, see perl -V for more detail)

Copyright 1987-2017, Larry Wall**

**Comment by adthrasher** [comment]
It appears that you have incompatible versions of Perl and GNU parallel. Try using Perl 5.10.1.

**Comment by JAYRJPT** [comment]
I have tried installing Perl 5.10.1 in my system but I am unable to do so. Only the latest version is getting installed without any error.
And the latest version of Perl is incompatible with GNU parallel.

**Comment by adthrasher** [comment]
Have you tried building and running via the included Dockerfile?

**Comment by JAYRJPT** [comment]
Yes, I have tried building the Dockerfile.
I got error while building Docker Image.
```
docker build -t stjude/cicero:0.2.0 .
error checking context: 'can't stat '/media/deepak/EXTRA/CICERO-master/dockerfiles''.
```

**Comment by adthrasher** [comment]
Do you have a file/directory/symlink named "dockerfiles" in your local directory? That isn't part of the distribution, but it appears that Docker is having issues accessing that.

**Comment by JAYRJPT** [comment]
Thanks for your suggestion. 
I successfully build the Docker image.
```
Successfully built 38d5438a8386
Successfully tagged stjude/cicero:0.2.0
```
However when I am invoking the CICERO wrapper using Docker I am getting the error.
`sudo docker run -v /home/deepak/Downloads/reference:/reference stjude/cicero:0.2.0 -n 20 -b /home/deepak/SJBALL020016_C27.bam -g GRCh37-lite -r /home/deepak/Downloads/reference:/reference -o /media/deepak/EXTRA/output`
```
docker: Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \"-b\": executable file not found in $PATH": unknown.
ERRO[0002] error waiting for container: context canceled 
```

**Comment by JAYRJPT** [comment]
Although Docker Image is build but the repository is not running, showing error.
I think that is why I am unable to invoke Cicero wrapper using Docker image.
Here my command with outputs
`sudo docker images`
```


REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
 stjude/cicero       0.2.0               38d5438a8386        8 hours ago         2.47GB
ubuntu              latest              74435f89ab78        9 days ago          73.9MB

```

`sudo docker run -it stjude/cicero`
```
Unable to find image 'stjude/cicero:latest' locally
docker: Error response from daemon: pull access denied for stjude/cicero, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
```

Please give your suggestion.
Thanks and regards

**Comment by adthrasher** [comment]
The build tags the image as "0.2.0". Attempting to run without a specific tag defaults to "latest", which does not exist in your repository. 

Using your original command line, you need to add appropriate mounts to get your inputs into the container. 

`sudo docker run -v /home/deepak/:/home/deepak/ -v /media/deepak/EXTRA/:/media/deepak/EXTRA/ stjude/cicero:0.2.0 Cicero.sh -n 20 -b /home/deepak/SJBALL020016_C27.bam -g GRCh37-lite -r /home/deepak/Downloads/reference -o /media/deepak/EXTRA/output`

**Comment by JAYRJPT** [comment]
Thanks for suggesting the command line which helped me to run the tool.
However I got some errors.
**Step 01 - 2020.06.29 18:05:49 - ExtractSClips
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.10.42254936.42299999.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.1.121360001.121399999.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.1.121500001.121535433.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.1.124535435.142529999.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.11.51600001.51644204.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.1.2586401.2613999.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.13.10001.15999999.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.14.10001.15999999.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.15.10001.16999999.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.16.34200001.35335800.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.16.38335802.46299999.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.18.10001.106499.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.18.15420001.15460897.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.18.18460899.18499999.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.19.24630001.24681781.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.19.27681783.27699999.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.20.26290001.26369568.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.20.29369570.29418999.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.21.10001.9409999.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.21.11190001.11288128.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.21.14288130.14337999.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.22.10001.12999999.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.2.89880001.90369999.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.2.90490001.91599999.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.6.57600001.58699999.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.6.58800001.58830165.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.7.58000001.58054330.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.7.61054332.61729999.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.9.50367680.66832999.cover has no soft clipped reads
CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.X.61632013.61679999.cover has no soft clipped reads
Step 02 - 2020.06.29 18:29:18 - Cicero
Step 03 - 2020.06.29 18:39:17 - Combine
Step 04 - 2020.06.29 18:39:17 - Annotate
Step 05 - 2020.06.29 18:51:30 - Filter
Killing blat server with pid 147
Killing   147 ?        00:04:01 gfServer**

**Comment by adthrasher** [comment]
Those messages are just informative. It does mean that large portions of the sample do not have soft clip reads in the BAM so there is no signal on which to detect structural variants. 

Can you check your output directory? Was a "final_fusions.txt" file produced? If not, can you share the contents of the log files in the output directory?

**Comment by JAYRJPT** [comment]
No the "final_fusions.txt" file is not produced.

**01_ExtractSClips.log**

Seq	Host	Starttime	JobRuntime	Send	Receive	Exitval	Signal	Command
7	:	1593453950.036	     0.185	0	0	0	0	extract_range.pl --ref_genome /home/deepak/Downloads/reference/Homo_sapiens/GRCh37-lite/FASTA/GRCh37-lite.fa -i /home/deepak/SJBALL020016_C27.bam -o CICERO_DATADIR/SJBALL020016_C27 -r 1:124535435-142529999 -l 100 -m 2 -min_sc_len 3 -c 10
14	:	1593453950.053	     0.229	0	0	0	0	extract_range.pl --ref_genome /home/deepak/Downloads/reference/Homo_sapiens/GRCh37-lite/FASTA/GRCh37-lite.fa -i /home/deepak/SJBALL020016_C27.bam -o CICERO_DATADIR/SJBALL020016_C27 -r 2:90490001-91599999 -l 100 -m 2 -min_sc_len 3 -c 10
2	:	1593453950.026	     0.259	0	0	0	0	extract_range.pl --ref_genome /home/deepak/Downloads/reference/Homo_sapiens/GRCh37-lite/FASTA/GRCh37-lite.fa -i /home/deepak/SJBALL020016_C27.bam -o CICERO_DATADIR/SJBALL020016_C27 -r 1:2586401-2613999 -l 100 -m 2 -min_sc_len 3 -c 10
6	:	1593453950.034	     0.255	0	0	0	0	extract_range.pl --ref_genome /home/deepak/Downloads/reference/Homo_sapiens/GRCh37-lite/FASTA/GRCh37-lite.fa -i /home/deepak/SJBALL020016_C27.bam -o CICERO_DATADIR/SJBALL020016_C27 -r 1:121500001-121535433 -l 100 -m 2 -min_sc_len 3 -c 10
.
.
.
51	:	1593454036.779	  1321.576	0	0	0	0	extract_range.pl --ref_genome /home/deepak/Downloads/reference/Homo_sapiens/GRCh37-lite/FASTA/GRCh37-lite.fa -i /home/deepak/SJBALL020016_C27.bam -o CICERO_DATADIR/SJBALL020016_C27 -r 14:19100001-107339539 -l 100 -m 2 -min_sc_len 3 -c 10

**02_Cicero.log**
Seq	Host	Starttime	JobRuntime	Send	Receive	Exitval	Signal	Command
18	:	1593455375.899	    20.613	0	0	0	0	Cicero.pl -i /home/deepak/SJBALL020016_C27.bam -o CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.116 -l 100 -genome GRCh37-lite -f CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.116.SC -c 10
4	:	1593455375.867	    20.868	0	0	0	0	Cicero.pl -i /home/deepak/SJBALL020016_C27.bam -o CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.103 -l 100 -genome GRCh37-lite -f CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.103.SC -c 10
.
.
.
200	:	1593455784.358	   173.098	0	0	0	0	Cicero.pl -i /home/deepak/SJBALL020016_C27.bam -o CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.69 -l 100 -genome GRCh37-lite -f CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.69.SC -c 10

**04_Annotate.log**
Seq	Host	Starttime	JobRuntime	Send	Receive	Exitval	Signal	Command
2	:	1593455957.690	    36.845	0	113	0	0	annotate.pl -i /home/deepak/SJBALL020016_C27.bam -o CICERO_DATADIR/SJBALL020016_C27 -genome GRCh37-lite -l 100 -s SJBALL020016_C27 -f CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.gene_info.txt -internal -c 10
1	:	1593455957.687	   732.573	0	113	0	0	annotate.pl -i /home/deepak/SJBALL020016_C27.bam -o CICERO_DATADIR/SJBALL020016_C27 -genome GRCh37-lite -l 100 -s SJBALL020016_C27 -f CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.gene_info.txt -j  -c 10

**gfServer.a31cda9f3743.1.log**

2020/06/29 18:03:59: info: gfServer version 35 on host localhost, port 9466  (2020-06-29 18:03)
2020/06/29 18:03:59: info: setting up untranslated index
2020/06/29 18:05:35: info: indexing complete
2020/06/29 18:05:35: info: Server ready for queries!

**Comment by JAYRJPT** [comment]
> Those messages are just informative. It does mean that large portions of the sample do not have soft clip reads in the BAM so there is no signal on which to detect structural variants.
> 
> Can you check your output directory? Was a "final_fusions.txt" file produced? If not, can you share the contents of the log files in the output directory?

Also My output directory contains lots of temporary files.

**Comment by adthrasher** [comment]
I don't see anything in the logs to indicate failure of any tasks. I'm also unable to reproduce the error on my end. 

Can you do `ls -l /media/deepak/EXTRA/output/CICERO_DATADIR/SJBALL020016_C27/*.txt` (assuming that is your output path) and share the results?

**Comment by JAYRJPT** [comment]
rw-r--r-- 1 root root  109087 Jun 30 00:21 /media/deepak/EXTRA/output/CICERO_DATADIR/SJBALL020016_C27/annotated.all.txt
-rw-r--r-- 1 root root  101719 Jun 30 00:21 /media/deepak/EXTRA/output/CICERO_DATADIR/SJBALL020016_C27/annotated.fusion.txt
-rw-r--r-- 1 root root    7368 Jun 30 00:09 /media/deepak/EXTRA/output/CICERO_DATADIR/SJBALL020016_C27/annotated.internal.txt
-rw-r--r-- 1 root root       0 Jun 30 00:09 /media/deepak/EXTRA/output/CICERO_DATADIR/SJBALL020016_C27/blacklist.new.fusions.txt
-rw-r--r-- 1 root root       0 Jun 30 00:09 /media/deepak/EXTRA/output/CICERO_DATADIR/SJBALL020016_C27/blacklist.new.internal.txt
-rw-r--r-- 1 root root   84215 Jun 30 00:26 /media/deepak/EXTRA/output/CICERO_DATADIR/SJBALL020016_C27/final_fusions.txt
-rw-r--r-- 1 root root     685 Jun 30 00:26 /media/deepak/EXTRA/output/CICERO_DATADIR/SJBALL020016_C27/final_internal.txt
-rw-r--r-- 1 root root 1339648 Jun 29 23:49 /media/deepak/EXTRA/output/CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.gene_info.txt
-rw-r--r-- 1 root root  638790 Jun 29 23:59 /media/deepak/EXTRA/output/CICERO_DATADIR/SJBALL020016_C27/SJBALL020016_C27.SC.txt
-rw-r--r-- 1 root root  191627 Jun 30 00:09 /media/deepak/EXTRA/output/CICERO_DATADIR/SJBALL020016_C27/unfiltered.fusion.txt
-rw-r--r-- 1 root root 1761609 Jun 30 00:09 /media/deepak/EXTRA/output/CICERO_DATADIR/SJBALL020016_C27/unfiltered.internal.txt

**Comment by JAYRJPT** [comment]
> -rw-r--r-- 1 root root 84215 Jun 30 00:26 /media/deepak/EXTRA/output/CICERO_DATADIR/SJBALL020016_C27/final_fusions.txt

I can see the final_fusion.txt file in this directory. Is this the file we should look for?
Thanks for your suggestion.

**Comment by liqingti** [comment]
Yes. final_fusions.txt is what you need. You also can visualize it by
loading it to https://proteinpaint.stjude.org/FusionEditor/

Thanks,
Liqing

**Comment by JAYRJPT** [comment]
I have issue in RNApeg also.
Should I raised a separate issue or should I post here?
